android/quest/src/main/java/org/smartregister/fhircore/quest/ui/questionnaire/QuestionnaireViewModel.kt

```diff
@@ -274,13 +304,37 @@ constructor(
 
       softDeleteResources(questionnaireConfig)
 
+      retireUsedQuestionnaireUniqueId()
+
       val idTypes =
         bundle.entry?.map { IdType(it.resource.resourceType.name, it.resource.logicalId) }
           ?: emptyList()
       onSuccessfulSubmission(idTypes, currentQuestionnaireResponse)
     }
   }
 
+  suspend fun retireUsedQuestionnaireUniqueId() {
+    if (uniqueIdResourcePair != null) {
+      val (id, resource) = uniqueIdResourcePair!!
+
+      // Update Group resource. Can be extended in future to support other resources
+      if (resource is Group) {
```

Comment:
@ellykits In this implementation, prepoulated IDs are assumed to be used and therefore marked as used on Questionnaire submission. Since it is possible to edit the ID in the questionnaire (if not read-only), it is possible that the ID submitted with the questionnaire isn't the same as the one that was originally prepopulated. 
We might have to confirm by checking from the `QuestionnaireResponse`

Reply:
Is this because the field is editable?

Reply:
Interesting, good catch. I agree, it makes sense to check the QR and only mark it as used if the ID in the QR matches the one we are expecting to be used.

It seems OK for whether it's editable or not to be an implementation decisions, as long as the implementer understands that is it is editable, the IDs can no longer be guaranteed unique, let's note this in the docs